services/provider/server/consumer.go

```diff
@@ -17,16 +16,11 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-var (
-	errTicketAlreadyExists = errors.New("onboarding ticket already used by another storageConsumer")
-)
-
 type ocsConsumerManager struct {
-	client       client.Client
-	namespace    string
-	nameByTicket map[string]string
-	nameByUID    map[types.UID]string
-	mutex        sync.RWMutex
+	client    client.Client
+	namespace string
+	nameByUID map[types.UID]string
+	mutex     sync.RWMutex
```

Comment:
One other question. Does the mutex make any more sense?
Are we trying to protect more then just access to the map and in that case isn't a sync map makes more sense

Reply:
I'd still want to stick w/ normal map w/ lock due to typing guarantees and the docs https://pkg.go.dev/sync#Map also says against it unless specialized in some some nature.
```
The Map type is specialized. Most code should use a plain Go map instead, with separate locking or coordination, for better type safety and to make it easier to maintain other invariants along with the map content.
```